lib/nats.js

```diff
@@ -1534,29 +1545,33 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
         msg = msg === undefined ? null : msg;
     }
 
+    if (opt_callback && typeof opt_callback !== 'function') {
+        throw (new NatsError(BAD_CALLBACK_MSG, BAD_CALLBACK));
+    }
```

Comment:
This guards against using the fourth argument as a callback when it is not a function.